src/future.rs

```diff
@@ -141,9 +139,10 @@ where
             Poll::Ready(result) => {
               inner.future_or_output = FutureOrOutput::Output(result.clone());
               drop(inner); // stop borrow_mut
-              let wakers = std::mem::take(
-                &mut *self.0.child_waker_state.wakers.borrow_mut(),
-              );
```

Comment:
I think maybe this one is the one that caused the issue? I just changed them all to use blocks to be sure because I always get bitten by this.